libavformat/dashenc.c

```diff
@@ -2341,10 +2357,12 @@ static int dash_write_packet(AVFormatContext *s, AVPacket *pkt)
         c->start_time_s = curr_time / 1000000;
 
         av_log(s, AV_LOG_INFO, "----------------------------------------\n");
-        rel_init_time = get_init_time(pkt);
-        if (rel_init_time <= 0) {
+        rel_init_time = get_init_time(s, pkt);
+        if (rel_init_time == 0) {
             av_log(s, AV_LOG_INFO, "Init time of pkt = 0, codec: %s. So skip optimising availabilityStartTime.\n", os->codec_str);
             c->availability_start_time_us = curr_time;
+        } else if (rel_init_time < 0) {
+            av_log(c, AV_LOG_INFO, "missing packet time ret: %"PRId64", codec: %s\n", rel_init_time, os->codec_str);
```

Comment:
Dont we need to set `c->availability_start_time_us` under this condition?

Reply:
Fixed